### PR TITLE
Space filling

### PR DESCRIPTION
**Pull request by jimbarrett27**

### CODE REVIEW

File: Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,561 @@
+{
```
Comment (naslundx) [inline Untitled.ipynb]: Hmm, maybe notebooks shouldn't be added to the repo (and in fact even added to .gitignore?)

File: wrdcld/font.py [hunk 2]
```diff
@@ -34,8 +34,8 @@ def draw_text(canvas, img, rectangle, word, font, rotate=False):
         text_draw.text((0, 0), word, font=font, fill=FONT_COLOR)
         rotated_text_image = text_image.rotate(90, expand=True)
         img.paste(rotated_text_image, rectangle.xy)
-        # canvas.rectangle(rectangle.corners)
+        # canvas.rectangle(rectangle.xyrb)
```
Comment (naslundx) [inline wrdcld/font.py:44]: I think if you merge my other PR this will simplify

File: pyproject.toml [hunk 3]
```diff
@@ -9,6 +9,7 @@ readme = "README.md"
 [tool.poetry.dependencies]
 python = "^3.9"
 pillow = "^10.4.0"
+numpy = "^2.0.1"
```
Comment (naslundx) [inline pyproject.toml:12]: As we talked about already - I think this is too heavyweight :)

File: wrdcld/rectangle.py [hunk 4]
```diff
@@ -47,9 +47,151 @@ def __repr__(self):
         return f"Rectangle(x={int(self.x)} y={int(self.y)} w={int(self.width)} h={int(self.height)})"
 
 
-def fill_remaining_space(
+def fill_remaining_space_horizontal(
```
Comment (naslundx) [inline wrdcld/rectangle.py:85]: Did you remove this in the previous commit, and re-add it here?

File: wrdcld/font.py [hunk 5]
```diff
@@ -26,8 +26,12 @@ def find_fontsize_for_width(width, word):
 
 
 def draw_text(canvas, img, rectangle, word, font, rotate=False):
+    """
+
```
Comment (naslundx) [inline wrdcld/font.py]: Good talk

File: wrdcld/rectangle.py (hunk 4 above)
Comment (jimbarrett27) [reply wrdcld/rectangle.py:84]: mayyyybe 

I had removed it, but then the filling around the words tended to make too many long narrow rectangles, so I decided to keep the old way and then only use the text filling in the text_rectangle

**Pull request merged**